### ISSUE TITLE
Fix multiprocessing version issues

### ISSUE THREAD

**Issue by jcass11** [AllenCellModeling/diffusive_distinguishability#1]
* Python version: 3.6

### Description

When 'multiprocessing' is used, all code functions fine. However, readthedocs fails to build because the only version it finds of multiprocessing is written for python2 and includes print statements with parentheses. 'multiprocess' is a fork of multiprocessing which is python3 compatible, but when imported in jupyter notebooks, it can't be found. Pip install as per PyPI (https://pypi.org/project/multiprocess/) fails. I want to find a fix that imports and runs, and is python3 compatible.

### What I Did

```
#When building readthedocs with multiprocessing:
Collecting multiprocessing
  Using cached https://files.pythonhosted.org/packages/b8/8a/38187040f36cec8f98968502992dca9b00cc5e88553e01884ba29cbe6aac/multiprocessing-2.6.2.1.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/private/var/folders/05/254ppq656dnfc4rl_nfv84sr0000gn/T/pip-install-9dzh3cfr/multiprocessing/setup.py", line 94
        print 'Macros:'
                      ^
    SyntaxError: Missing parentheses in call to 'print'. Did you mean print(int 'Macros:')?


#When installing multiprocess:
juliec@n84:~$ pip install multiprocess
Collecting multiprocess
  DEPRECATION: Failed to find 'multiprocess' at https://artifactory.corp.alleninstitute.org/artifactory/api/pypi/pypi-virtual/simple/multiprocess/. It is suggested to upgrade your index to support normalized names as the name in /simple/{name}.
  Cannot fetch index base URL https://artifactory.corp.alleninstitute.org/artifactory/api/pypi/pypi-virtual/simple/
  Could not find a version that satisfies the requirement multiprocess (from versions: )
No matching distribution found for multiprocess
juliec@n84:~$
```

**Comment by evamaxfield** [comment]
I am not sure how I received a notification for this but I am happy I did.

Do you have a moment for me to tell you about our savior `concurrent.futures`?

https://docs.python.org/3/library/concurrent.futures.html

In short: it is a wrapper around the multiprocessing library with niceties so commonly used that `concurrent` is a standard library in python 3.6+

To switch from `multiprocessing` to `concurrent` it is like this:
```python
from concurrent.futures import ProcessPoolExecutor

with ProcessPoolExecutor() as exe:
   results = list(exe.map(your_func, your_iterable))
```

If this doesn't matter, it doesn't matter. But if you want me to explain the niceties tomorrow: I would be happy to.

**Comment by jcass11** [comment]
Oh that’s awesome! I have no idea why you got a notification either but I’m so glad you did and have a possible answer! Thanks I’ll talk to you about it tomorrow.

Sent from my iPhone

On Apr 29, 2019, at 6:27 PM, JacksonMaxfield <notifications@github.com<mailto:notifications@github.com>> wrote:


I am not sure how I received a notification for this but I am happy I did.

Do you have a moment for me to tell you about our savior concurrent.futures?

https://docs.python.org/3/library/concurrent.futures.html<https://nam05.safelinks.protection.outlook.com/?url=https%3A%2F%2Fdocs.python.org%2F3%2Flibrary%2Fconcurrent.futures.html&data=02%7C01%7C%7Ca3dbcc48e3b147de65a108d6cd0b117f%7C32669cd6737f4b398bddd6951120d3fc%7C0%7C0%7C636921844774119246&sdata=fQj0pzbaRQNkb9Mb8M5c3OcTwArJJDsm88sNMfmB%2BOM%3D&reserved=0>

In short: it is a wrapper around the multiprocessing library with niceties so commonly used that concurrent is a standard library in python 3.6+

To switch from multiprocessing to concurrent it is like this:

from concurrent.futures import ProcessPoolExecutor

with ProcessPoolExecutor() as exe:
   results = list(exe.map(your_func, your_iterable))


If this doesn't matter, it doesn't matter. But if you want me to explain the niceties tomorrow: I would be happy to.

**Comment by jcass11** [comment]
Switched from multiprocessing to concurrent.futures based on Jackson's recommendation.

Code now runs as expected and readthedocs builds successfully.
